docs/grammar/SolidityParser.g4

```diff
@@ -273,7 +273,10 @@ locals [boolean constantnessSet = false, boolean visibilitySet = false, boolean
 		| {!$overrideSpecifierSet}? overrideSpecifier {$overrideSpecifierSet = true;}
 		| {!$constantnessSet}? Immutable {$constantnessSet = true;}
 	)*
-	name=identifier
+	(
+		location=Transient name=identifier
+		| name=identifier
```

Comment:
IIRC the operator precedence here is such that this will interpreted as `location=Transient (name=identifier | name=identifier)`, which is not what we want. I think you'd need to add parentheses for it to work as intended.

Also the `name` variable will end up being a different field depending on whether we have `transient` or not. See my previous discussion with Daniel about a similar issue in grammar for literal suffixes: https://github.com/ethereum/solidity/pull/12656#discussion_r1169003951

What was actually wrong with adding `Transient` to the rule above along with other specifiers?

Reply:
I think that this bit still has that issue, doesn't it?
```
	type=typeName
	(
		transientLocation=Transient name=identifier?
		| location=dataLocation? name=identifier
	);
```

Reply:
After discussing it with @ekpyron, I think this is probably fine after all. `A B | C D` should be equivalent to `(A B) | (C D)` in ANTLR. We do have some rules in the grammar that to go out of their way to parenthesize this but that's probably unnecessary. It would not hurt to double check, but if that was wrong, then a lot of rules (e.g. `expression`) would be broken and tests should have detected that.